Isabelle/Chi_Calculus/Relaying.thy

```diff
@@ -13,32 +13,46 @@ lemma early_multi_receive_redundancy:
 
 lemma shortcut_redundancy:
   shows "a \<rightarrow> b \<parallel> b \<rightarrow> c \<parallel> a \<rightarrow> c \<approx>\<^sub>\<flat> a \<rightarrow> b \<parallel> b \<rightarrow> c"
-  by (fact early_multi_receive_redundancy)
+  using unidirectional_bridge_def and early_multi_receive_redundancy by simp
 
 lemma loop_redundancy_under_duploss:
   shows "\<currency>\<^sup>*a \<parallel> a \<rightarrow> a \<approx>\<^sub>\<flat> \<currency>\<^sup>*a"
   sorry
 
-abbreviation bidirectional_bridge :: "[chan, chan] \<Rightarrow> process" (infix "\<leftrightarrow>" 100) where
+definition bidirectional_bridge :: "[chan, chan] \<Rightarrow> process" (infix "\<leftrightarrow>" 100) where
   "a \<leftrightarrow> b \<equiv> a \<rightarrow> b \<parallel> b \<rightarrow> a"
 
 lemma bidirectional_bridge_commutativity: "a \<leftrightarrow> b \<sim>\<^sub>\<flat> b \<leftrightarrow> a"
-  by (simp add: parallel_commutativity)
+  by (simp add: parallel_commutativity bidirectional_bridge_def)
```

Comment:
What I said [above](https://github.com/input-output-hk/fm-ouroboros/pull/145/files#r311021714) applies also here. What’s different here is that we already used `simp` before switching from `abbreviation` to `definition`, so that applying the defining equation separately via `unfolding` might seem like overkill. However, `simp` is not necessary to prove this statement; a simple `fact` would be enough. Therefore, we’ll move away from using `simp` here anyhow when resolving #144.